### PR TITLE
Add support for Junit5 tests and fix CI that has been broken

### PR DESCRIPTION
**Pull request by lhotari**
### Motivation

Add support for JUnit 5 tests so that tests can be written in modern ways. Parameterized tests in Junit 4 are painful.

### Changes

- upgrade surefire plugin
- replace <forkMode>always</forkMode> with supported way in newer maven-surefire-plugin
- maven-failsafe-plugin version must match maven-surefire-plugin version
- remove invalid test case. Log4j is in the classpath for other reasons checked by running "mvn -pl tests/integration-tests-utils/pom.xml test -X" also verified by debugging that it's not in the system classpath, it is loaded by the MavenClassLoader
- add assertj-core for fluent assertions in tests

### Addition Context

While working on this, it was noticed that apache/bookkeeper maven build hasn't been running all tests in GitHub Actions CI. Mailing list thread: https://lists.apache.org/thread/5c2yr772v6o0kcv717g0csgv44w6wthc

This PR is fixing the broken CI too since broken tests started getting run when Junit 5 test support was added to the maven build.

### CODE REVIEW

**Comment by lhotari** [comment]
@merlimat @hangc0276 @BewareMyPower please review

**Comment by lhotari** [comment]
@nicoloboschi @eolivelli please review

**Review by eolivelli — APPROVED** [review]
+1
thanks for taking care of the testing framework

**Review by BewareMyPower — APPROVED** [review]

**Review by nicoloboschi — APPROVED** [review]

**Review by hangc0276 — APPROVED** [review]

**Comment by hangc0276** [comment]
Good job!

File: circe-checksum/pom.xml [hunk 1]
```diff
@@ -80,8 +80,28 @@
...
+              <narSystemPackage>com.scurrilous.circe.checksum</narSystemPackage>
+            </library>
+          </libraries>
+          <cpp>
+            <optionSet>${nar.cpp.optionSet}</optionSet>
+            <exceptions>false</exceptions>
+            <rtti>false</rtti>
+            <optimize>full</optimize>
+            <includes>
+              <include>**/*.cpp</include>
```
Comment (lhotari) [inline circe-checksum/pom.xml:99]: This is the actual fix

**Review by eolivelli — APPROVED** [review]
Great work

thank you very much

**Comment by eolivelli** [comment]
merging as soon as CI passes

**Comment by lhotari** [comment]
> merging as soon as CI passes

@eolivelli Integration tests have been ignored in apache/bookkeeper CI for a very long time. Here's some evidence: https://github.com/apache/bookkeeper/pull/3675#issuecomment-1933485215 . I had to fix a bug in `bkctl ledger simpletest" `and it appears that it was already merged as invalid in November 2022. 😱

**Pull request merged**